python/ray/train/v2/tests/test_v2_api.py

```diff
@@ -122,6 +124,50 @@ def test_train_v2_import(monkeypatch, env_v2_enabled):
         assert Result is not ResultV2
 
 
+def test_exceptions_are_picklable():
+    """Test that TrainingFailedError and ControllerError are picklable."""
+    # Test TrainingFailedError
+    worker_failures = {
+        0: ValueError("worker 0 failed"),
+        1: RuntimeError("worker 1 failed"),
+    }
+    training_error = TrainingFailedError(
+        "Training failed on multiple workers", worker_failures
+    )
+
+    # Test pickle/unpickle for TrainingFailedError
+    pickled_training_error = pickle.dumps(training_error)
+    unpickled_training_error = pickle.loads(pickled_training_error)
+
+    # Verify attributes are preserved
+    assert str(unpickled_training_error) == str(training_error)
+    assert (
+        unpickled_training_error._error_message == "Training failed on multiple workers"
+    )
+
+    # Compare worker failures by type and message since Exception objects don't implement __eq__
+    assert len(unpickled_training_error.worker_failures) == len(worker_failures)
+    for rank, original_exc in worker_failures.items():
+        unpickled_exc = unpickled_training_error.worker_failures[rank]
+        assert type(unpickled_exc) is type(original_exc)
+        assert str(unpickled_exc) == str(original_exc)
```

Comment:
no need for these assertions -- we just care about the top-level picklability. we can assume that the nested exceptions in this dict are picklable. (we should test pickling/unpickling all the RayTrainError subclasses in a followup PR).